command/connect/enable/connect_enable.go

```diff
@@ -0,0 +1,101 @@
+package enable
+
+import (
+	"flag"
+	"fmt"
+	"io"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/command/flags"
+	"github.com/mitchellh/cli"
+)
+
+func New(ui cli.Ui) *cmd {
+	c := &cmd{UI: ui}
+	c.init()
+	return c
+}
+
+type cmd struct {
+	UI    cli.Ui
+	flags *flag.FlagSet
+	http  *flags.HTTPFlags
+	help  string
+
+	service      string
+	protocol     string
+	sidecarProxy bool
+
+	testStdin io.Reader
+}
+
+func (c *cmd) init() {
+	c.flags = flag.NewFlagSet("", flag.ContinueOnError)
+	c.http = &flags.HTTPFlags{}
+	c.flags.BoolVar(&c.sidecarProxy, "sidecar-proxy", false, "Whether the service should have a Sidecar Proxy by default")
+	c.flags.StringVar(&c.service, "service", "", "The service to enable connect for")
+	c.flags.StringVar(&c.protocol, "protocol", "", "The protocol spoken by the service")
+	flags.Merge(c.flags, c.http.ClientFlags())
+	flags.Merge(c.flags, c.http.ServerFlags())
+	c.help = flags.Usage(help, c.flags)
+}
+
+func (c *cmd) Run(args []string) int {
+	if err := c.flags.Parse(args); err != nil {
+		return 1
+	}
+
+	if c.service == "" {
+		c.UI.Error("Must specify the -service parameter")
+		return 1
+	}
+
+	entry := &api.ServiceConfigEntry{
+		Kind:     api.ServiceDefaults,
+		Name:     c.service,
+		Protocol: c.protocol,
+		Connect: api.ConnectConfiguration{
+			SidecarProxy: c.sidecarProxy,
+		},
+	}
```

Comment:
This should probably take the existing config for the service (if there is one) and modify it based on the flags set. Right now this is probably fine since there's only a couple options but it'd be easy to overwrite a service's config this way.

Reply:
Will open an issue to update this in the future. We should probably do a -update flag or something but that will be more important for updating the upstreams.